### PR TITLE
[Microsoft.Workoads] Adding a new Api-version 2023-10-01-preview

### PR DESCRIPTION
**Pull request by sushantyadav-msft**
# ARM (Control Plane) API Specification Update Pull Request 

## PR review workflow diagram

Please understand this diagram before proceeding. It explains how to get your PR approved & merged.

diagram

[1] [ARM review queue] (for **merge** queues, see [4])  
The PRs are processed by time opened, ascending. Your PR may show up on 2nd or later page. 
If you addressed Step 1 from the diagram and your PR is not showing up in the queue, ensure the label `ARMChangesRequested` 
is removed from your PR. This should cause the label `WaitForARMFeedback` to be added.  
[2] https://aka.ms/azsdk/support/specreview-channel  
[3] [List of SDK breaking changes approvers] in pinned Teams announcement  
[4] [public repo merge queue], [private repo merge queue] (for **ARM review** queue, [1])

If you need further help with anything, see `Getting help` section below.

## Purpose of this PR

This is to add a new api version 2023-10-01-preview for Microsoft.workloads/sapvirtualinstances resource type. This is already present in the private repo.
**Couldn't use the Open API hub tool because we're moving to a new folder structure where each RT under the RP will have their own versioning.**
The linter check getting failed is a false positive and seems like a bug in the spec validation automation.
Requesting to review the changes in the PR.

  - [x] New API version. (Such PR should have been generated with [OpenAPI Hub](https://aka.ms/openapihub), per [this wiki doc](https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/208/OpenAPI-Hub-Adding-new-API-version).)
  - [ ] Update existing version for a new feature. (This is applicable only when you are revising a private preview API version.)
  - [ ] Update existing version to fix swagger quality issues in S360.
  - [ ] Other, please clarify:
    - _edit this with your clarification_

## Due diligence checklist

To merge this PR, you **must** go through the following checklist and confirm you understood 
and followed the instructions by checking all the boxes:

- [x] I confirm this PR is modifying Azure Resource Manager (ARM) related specifications, and not data plane related specifications.
- [ ] I have reviewed following [Resource Provider guidelines](https://aka.ms/rpguidelines), including
  [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and
  [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md) (estimated time: 4 hours).  
  I understand this is required before I can proceed to Step 2, "ARM Review", for this PR.

### Breaking changes review (Step 1)

- If the automation determines you have breaking changes, i.e. Step 1 from the diagram applies to you,
  you must follow the [breaking changes process].  
  **IMPORTANT** This applies even if:
  - The tool fails while it shouldn't, e.g. due to runtime exception, or incorrect detection of breaking changes.
  - You believe there is no need for you to request breaking change approval, for any reason. 
    Such claims must be reviewed, and the process is the same.

### ARM API changes review (Step 2)

- If this PR is in purview of ARM review then automation will add the `ARMReview` label.
- If you want to force ARM review, add the label yourself.
- Proceed according to the diagram at the top of this comment.

## Viewing API changes

For convenient view of the API changes made by this PR, refer to the URLs provided in the table 
in the `Generated ApiView` comment added to this PR. You can use ApiView to show API versions diff. 

## Suppressing failures

If one or multiple validation error/warning suppression(s) is detected in your PR, please follow the 
[Swagger-Suppression-Process](https://aka.ms/azsdk/pr-suppressions) 
to get approval.

## Getting help

- For general PR approval workflow, see the diagram at the top of this comment.
- For guidance on fixing this PR CI check failures, see the hyperlinks provided in given failure 
  and https://aka.ms/ci-fix.
- If the help provided by the previous points is not enough, post to https://aka.ms/azsdk/support/specreview-channel and link to this PR.

[ARM review queue]: https://github.com/search?q=org%3AAzure+is%3Apr+is%3Aopen+label%3AWaitForARMFeedback+-label%3AIDCDevDiv++draft%3Afalse+sort%3Acreated-asc+&type=pullrequests
[List of SDK breaking changes approvers]: https://teams.microsoft.com/l/message/19:0351f5f9404446e4b4fd4eaf2c27448d@thread.skype/1689115217750?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=3e17dcb0-4257-4a30-b843-77f47f1d4121&parentMessageId=1689115217750&teamName=Azure%20SDK&channelName=API%20Spec%20Review&createdTime=1689115217750
[public repo merge queue]: https://github.com/Azure/azure-rest-api-specs/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+draft%3Afalse+sort%3Acreated-asc
[private repo merge queue]: https://github.com/Azure/azure-rest-api-specs-pr/pulls?q=is%3Aopen+is%3Apr+label%3AMergeRequested+-label%3AApproved-OkToMerge+draft%3Afalse+sort%3Acreated-asc
[breaking changes process]: https://eng.ms/docs/cloud-ai-platform/azure-core/azure-core-pm-and-design/trusted-platform-pm-karimb/service-lifecycle-and-actions-team/service-lifecycle-actions-team/apex/media/launchingproductbreakingchanges#breaking-change-process-1

### CODE REVIEW

**Comment by mikekistler** [comment]
Breaking changes previously reviewed and approved in https://github.com/Azure/azure-rest-api-specs-pr/pull/14752

**Comment by sushantyadav-msft** [comment]
The linter check getting failed is a false positive and seems like a bug in the spec validation automation.

**Comment by raosuhas** [comment]
> The linter check getting failed is a false positive and seems like a bug in the spec validation automation.

Ack. Please ignore the PropertiesTypeObjectNoDefinition for the userassigned definition. I have flagged it as a bug

**Comment by raosuhas** [comment]
> > The linter check getting failed is a false positive and seems like a bug in the spec validation automation.
> 
> Ack. Please ignore the PropertiesTypeObjectNoDefinition for the userassigned definition. I have flagged it as a bug

Can you please fix the warnings meanwhile : [PostOperationIdContainsUrlVerb](https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-operation-id-contains-url-verb.md)

**Comment by raosuhas** [comment]
> > > The linter check getting failed is a false positive and seems like a bug in the spec validation automation.
> > 
> > 
> > Ack. Please ignore the PropertiesTypeObjectNoDefinition for the userassigned definition. I have flagged it as a bug
> 
> Can you please fix the warnings meanwhile : [PostOperationIdContainsUrlVerb](https://github.com/Azure/azure-openapi-validator/blob/main/docs/post-operation-id-contains-url-verb.md)

I just see that all these are also getting flagged for existing operations. Ignore this one as well. I will follow up on them

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/examples/sapapplicationinstances/SAPApplicationServerInstances_Create.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
+{
+  "parameters": {
+    "subscriptionId": "6d875e77-e412-4d7d-9af4-8895278b4443",
+    "resourceGroupName": "test-rg",
+    "sapVirtualInstanceName": "X00",
+    "applicationInstanceName": "app01",
+    "api-version": "2023-10-01-preview",
+    "body": {
+      "location": "westcentralus",
```
Comment (kalyaninamuduri) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/examples/sapapplicationinstances/SAPApplicationServerInstances_Create.json:9]: Remove westcentralus from all examples as that is no longer a supported region for ACSS.

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json [hunk 2]
```diff
@@ -0,0 +1,4455 @@
...
+          "final-state-via": "location"
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "Stop the SAP Application Server Instance": {
+            "$ref": "./examples/sapapplicationinstances/SAPApplicationServerInstances_StopInstance.json"
+          },
+          "Stop the SAP Application Server Instance and it's infrastructure.": {
```
Comment (kalyaninamuduri) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:1818]: Rephrase this as 
Stop the SAP Application Server Instance and the Virtual machine.

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json [hunk 3]
```diff
@@ -0,0 +1,4455 @@
...
+        "tags": [
+          "SAPApplicationServerInstances"
+        ],
+        "operationId": "SAPApplicationServerInstances_Update",
+        "x-ms-examples": {
+          "SAPApplicationServerInstances_Update": {
+            "$ref": "./examples/sapapplicationinstances/SAPApplicationServerInstances_Update.json"
+          }
+        },
+        "description": "Puts the SAP Application Server Instance resource.",
```
Comment (kalyaninamuduri) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:1301]: Rephrase the description as
Updates the SAP Application server instance resource. This can be used to update tags on the resource.

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json [hunk 4]
```diff
@@ -0,0 +1,4455 @@
...
+        "tags": [
+          "SAPDatabaseInstances"
+        ],
+        "operationId": "SAPDatabaseInstances_Update",
+        "x-ms-examples": {
+          "SAPDatabaseInstances_Update": {
+            "$ref": "./examples/sapdatabaseinstances/SAPDatabaseInstances_Update.json"
+          }
+        },
+        "description": "Updates the Database resource.",
```
Comment (kalyaninamuduri) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:1034]: Rephrase this as 
Updates the Database instance resource. This can be used to update tags on the resource.

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json [hunk 5]
```diff
@@ -0,0 +1,4455 @@
...
+        "operationId": "SAPVirtualInstances_Start",
+        "x-ms-long-running-operation": true,
+        "x-ms-long-running-operation-options": {
+          "final-state-via": "location"
+        },
+        "produces": [
+          "application/json"
+        ],
+        "x-ms-examples": {
+          "SAPVirtualInstances_Start": {
```
Comment (kalyaninamuduri) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:1461]: Rename the heading as 
Start the SAP application, which includes central services instance and the application server instances.

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json [hunk 6]
```diff
@@ -0,0 +1,4455 @@
...
+            "$ref": "./examples/sapvirtualinstances/SAPVirtualInstances_Start.json"
+          },
+          "SAPVirtualInstances_Start_WithInfraOperations": {
```
Comment (kalyaninamuduri) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:1464]: Rephrase this as 
Start the virtual machines and the SAP system

**Review by kalyaninamuduri — CHANGES_REQUESTED** [review]
Suggested changes

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/examples/sapapplicationinstances/SAPApplicationServerInstances_Create.json (hunk 1 above)
Comment (sushantyadav-msft) [reply specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/examples/sapapplicationinstances/SAPApplicationServerInstances_Create.json]: done

**Comment by ms-zhenhua** [comment]
please remove 200 from DELETE to fix lint error in https://github.com/Azure/azure-rest-api-specs/pull/25896/checks?check_run_id=17558768092. Then readd MergeRequest label

**Review by ms-zhenhua — APPROVED** [review]

File: specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json [hunk 7]
```diff
@@ -0,0 +1,4452 @@
...
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Workloads/locations/{location}/sapVirtualInstanceMetadata/default/getSizingRecommendations": {
+      "post": {
+        "tags": [
+          "SAPRecommendations"
+        ],
+        "description": "Get SAP sizing recommendations by providing input SAPS for application tier and memory required for database tier",
+        "operationId": "SAPSizingRecommendations",
```
Comment (kazrael2119) [inline specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:43]: follow https://github.com/Azure/azure-openapi-validator/blob/main/docs/operation-id.md#output-message,
I don't think this operation ID is valid here

Comment (MaryGao) [reply specification/workloads/resource-manager/Microsoft.Workloads/SAPVirtualInstance/preview/2023-10-01-preview/SAPVirtualInstance.json:43]: The operationId seems wired to me, but it is released in previous version. @weidongxu-microsoft how do you think?
